server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -332,11 +332,13 @@ public PublicationContext newPublicationContext(
         boolean isRemotePublicationEnabled,
         PersistedStateRegistry persistedStateRegistry
     ) {
-        final PublicationContext publicationContext = new PublicationContext(
-            clusterChangedEvent,
-            isRemotePublicationEnabled,
-            persistedStateRegistry
-        );
+        if (isRemotePublicationEnabled == true) {
+            if (validateRemotePublicationOnAllNodes(clusterChangedEvent.state().nodes()) == true) {
```

Comment:
this check will be triggered with every state update and would be un-necessary unless nodes are joining the cluster. Also once cluster moves to remote publication if one node joins with state publication disabled would it be allowed? It shouldn't be. Once a cluster starts doing remote publishing, it shouldn't go back and forth with local/remote publication dynamically.

Reply:
What if the user wants to disable the remote publication. New nodes will join with remote publication disabled. At this point we should stop remote publishing, right ?

Reply:
ideally remote publishing should be controlled by cluster manager instead of individual nodes. CM decides when to enable/ disable using the setting. This min compatibility check is to ensure along with setting, during upgrades, it can smoothly enable it without triggering downtime but it is not an expectation during regular operation of a running cluster.